### PR TITLE
W-16124800 kafka-confluence-configuration-kt

### PR DESCRIPTION
**Pull request by kevintroller**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+<kafka:consumer-plaintext-connection groupId="JenkinsTest">
+<kafka:bootstrap-servers >
+	<kafka:bootstrap-server value="ec2-3-226-204-173.compute-1.amazonaws.com:9092" />
+	</kafka:bootstrap-servers>
+<kafka:additional-properties >
+  <kafka:additional-property key="key.deserializer" value="’io.confluent.kafka.serializers.KafkaAvroDeserializer” />
+    <kafka:additional-property key="value.deserializer value="io.confluent.kafka.serializers.KafkaAvroDeserializer" />
+       <kafka:additional-property key="schema.registry.url" value="https://example.us-east-2.aws.confluent.cloud" />
+      <kafka:additional-property key="basic.auth.credentials.source" value="USER_INFO" />
+      <kafka:additional-property key="basic.auth.user.info" value="HK6N653LCZCATBEZ:3swV2Zw6jLNODpdwAMqCB6f0+7DNUko/VMZc2w1v7C3APdRZn4PlNz7KH/hxqAdh" />
```
Comment (acurci-at-mulesoft) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]: This value should be an example tests, like: _**** PUT YOUR CREDENTIALS HERE ****_

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+<kafka:producer-plaintext-connection groupId="JenkinsTest">
+<kafka:bootstrap-servers >
+		<kafka:bootstrap-server value="ec2-3-226-204-173.compute-1.amazonaws.com:9092" />
+	</kafka:bootstrap-servers>
+<kafka:additional-properties >
+  <kafka:additional-property key="key.serializer" value="’io.confluent.kafka.serializers.KafkaAvroSerializer” />
+       <kafka:additional-property key="value.serializer value="io.confluent.kafka.serializers.KafkaAvroSerializer" />
+       <kafka:additional-property key="schema.registry.url" value="https://example.us-east-2.aws.confluent.cloud" />
+      <kafka:additional-property key="basic.auth.credentials.source" value="USER_INFO" />
+      <kafka:additional-property key="basic.auth.user.info" value="HK6N653LCZCATBEZ:3swV2Zw6jLNODpdwAMqCB6f0+7DNUko/VMZc2w1v7C3APdRZn4PlNz7KH/hxqAdh" />
```
Comment (acurci-at-mulesoft) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]: Idem

**Review by acurci-at-mulesoft — CHANGES_REQUESTED** [review]
I think the places that says:
`<kafka:additional-property key="basic.auth.user.info" value="HK6N653LCZCATBEZ:3swV2Zw6jLNODpdwAMqCB6f0+7DNUko/VMZc2w1v7C3APdRZn4PlNz7KH/hxqAdh" />`
should be something like:
`<kafka:additional-property key="basic.auth.user.info" value="*** YOUR CREDENTIALS ***" />`

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 3]
```diff
@@ -0,0 +1,180 @@
+= Integrate Confluent Schema Registry with Kafka Connector - Example 
+
+The following example shows how to configure the Confluent Schema Registry using Kafka Connector. 
+
+Imagine you are developing a Mule 4 app that processes streaming data from various sources. To ensure the data conforms to defined schemas and to manage schema evolution effectively, you integrate Confluent Schema Registry with your Kafka Connector. This setup is crucial for developers who need to enforce data integrity and compatibility across different versions of data producers and consumers.
+
+The configuration consits of two parts:
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
The configuration consists of two parts:
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+* Configure the Confluent Schema Registry in the Kakfa Connector connection using the *Additional Properties* field  (either for the *Consumer* or *Producer* operation).
+* Configure the Kafka Avro dependency in the POM file of the Mule app.
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
* Configuring the Confluent Schema Registry in the Kakfa Connector connection using the *Additional Properties* field (either for the *Consumer* or *Producer* operation)
* Configuring the Kafka Avro dependency in the POM file of the Mule app
```
A small rewording to align better with the previous introductory sentence.

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+The following example shows how to configure the Confluent Schema Registry using Kafka Connector. 
+
+Imagine you are developing a Mule 4 app that processes streaming data from various sources. To ensure the data conforms to defined schemas and to manage schema evolution effectively, you integrate Confluent Schema Registry with your Kafka Connector. This setup is crucial for developers who need to enforce data integrity and compatibility across different versions of data producers and consumers.
+
+The configuration consits of two parts:
+
+* Configure the Confluent Schema Registry in the Kakfa Connector connection using the *Additional Properties* field  (either for the *Consumer* or *Producer* operation).
+* Configure the Kafka Avro dependency in the POM file of the Mule app. 
+
+== Configure Kafka Connection
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]: Is this the same as the first part mentioned before? If so, I'd update the title to be more similar: 

== Configure the Confluent Schema Registry in the Kakfa Connector connection

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+Use the *Additional Properties* field to configure the Confluent Schema Registry.
+
+=== Configuration for Consumer Operation
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
=== Configure the Consumer Operation
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+
+To configure the *Consumer* operation in Anypoint Studio, follow these steps:
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
To configure the *Consumer* operation in Anypoint Studio:
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+
+. Open your Mule project in Anypoint Studio and navigate to the Global Elements tab.
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
. Open your Mule project in Anypoint Studio and navigate to the *Global Elements* tab.
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+. Click *Create*.
+. In the search box, type `kafka`.
+. Select *Apache Kafka Consumer configuration*.
+. Click *OK*.
+. In the configuration screen under connection, select *Advanced*.
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc:25]:
```suggestion
. In the configuration screen under *connection*, select *Advanced*.
```
Assuming "connection" is a UI element.

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+     <kafka:additional-property key="value.subject.name.strategy" value="io.confluent.kafka.serializers.subject.TopicNameStrategy" />	
+        </kafka:additional-properties>
+        <kafka:topic-patterns >
+		<kafka:topic-pattern value="InfraTopic" />
+	    </kafka:topic-patterns>
+        </kafka:consumer-plaintext-connection>
+	    </kafka:consumer-config>
+----
+
+=== Configuration for Producer Operation
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
=== Configure the Producer Operation
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+
+For the *Producer* configuration the difference is to delete `key.deserializer` and `value.deserializer` and replace it with `key.serializer` and `value.serializer`. To configure the *Producer* operation in Anypoint Studio, follow these steps:
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
For the *Producer* operation, you replace `key.deserializer` and `value.deserializer` with `key.serializer` and `value.serializer`. To configure the *Producer* operation in Anypoint Studio:
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+			
+</kafka:additional-properties>
+<kafka:topic-patterns >
+		<kafka:topic-pattern value="InfraTopic" />
+	</kafka:topic-patterns>
+</kafka:producer-plaintext-connection>
+	</kafka:producer-config>
+----
+
+== Configure POM file in Mule App
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
== Configure the Kafka Avro dependency in the POM File
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+
+In the POM file of the Mule app, you configure the Kafka Avro dependency that contains the class that you use to serialize and deserialize. You must update the `<dependencies>`, `<plugin>`, and `<repository>` sections of the POM file.
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
In the POM file of the Mule app, configure the Kafka Avro dependency that contains the class to use to serialize and deserialize. You must update the `<dependencies>`, `<plugin>`, and `<repository>` sections of the POM file.
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+
+=== Configure <dependencies> Section
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
=== Configure the <dependencies> Section
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+<dependencies>	
+        <dependency>
+            <groupId>io.confluent</groupId>
+            <artifactId>kafka-avro-serializer</artifactId>
+            <version>7.4.0</version>
+        </dependency>
+</dependencies>	
+----
+
+=== Configure <plugin> Section
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
=== Configure the <plugin> Section
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+				<groupId>io.confluent</groupId>
+				<artifactId>kafka-avro-serializer</artifactId>
+			</sharedLibrary>
+		</sharedLibraries>
+	<classifier>mule-application</classifier>
+	</configuration>
+</plugin>
+----
+
+Then, set the `avro-maven-plugin` plugin. This dependency from the org.apache.avro group is a plugin for Apache Maven that facilitates the handling of Avro schema files in Java projects. The plugin automatically generates Java classes from Avro schemas (.avsc), protocols (.avpr), and IDL files (.avdl). This simplifies development by eliminating the need to manually write Java classes that represent Avro schemas, ensuring that the Java code is always synchronized with the Avro schemas.
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
Then, set the `avro-maven-plugin` plugin. This dependency from the `org.apache.avro` group is a plugin for Apache Maven that facilitates handling Avro schema files in Java projects. The plugin automatically generates Java classes from Avro schemas (.avsc), protocols (.avpr), and IDL files (.avdl). This process simplifies development by eliminating the need to manually write Java classes that represent Avro schemas, ensuring that the Java code is always synchronized with the Avro schemas.
```

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+			</goals>
+			<configuration>							
+            <sourceDirectory>${project.basedir}/PATH_TO_SCHEMAS</sourceDirectory>
+			</configuration>
+		</execution>
+	</executions>
+</plugin>
+----
+
+== Configure <repository> Section
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
== Configure the <repository> Section
```

**Review by Cristian-Venticinque — COMMENTED** [review]
Please address comments.

File: kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc [hunk 18]
```diff
@@ -0,0 +1,180 @@
...
+The following example shows how to configure the Confluent Schema Registry using Kafka Connector. 
+
+Imagine you are developing a Mule 4 app that processes streaming data from various sources. To ensure the data conforms to defined schemas and to manage schema evolution effectively, you integrate Confluent Schema Registry with your Kafka Connector. This setup is crucial for developers who need to enforce data integrity and compatibility across different versions of data producers and consumers.
+
+The configuration consists of two parts:
+
+* Configuring the Confluent Schema Registry in the Kakfa Connector connection using the *Additional Properties* field (either for the *Consumer* or *Producer* operation)
+* Configuring the Kafka Avro dependency in the POM file of the Mule app
+
+== Configure the Confluent Schema Registry in the Kakfa Connector connection
```
Comment (Cristian-Venticinque) [inline kafka/4.8/modules/ROOT/pages/kafka-connector-configure-confluent-example.adoc]:
```suggestion
== Configure the Confluent Schema Registry in the Kakfa Connector Connection
```

**Review by Cristian-Venticinque — DISMISSED** [review]

**Review by Cristian-Venticinque — DISMISSED** [review]

**Review by Cristian-Venticinque — APPROVED** [review]

**Review by acurci-at-mulesoft — APPROVED** [review]

**Pull request merged**